src/main/java/com/rtb/projectmanagementtool/user/UserProfileServlet.java

```diff
@@ -98,24 +108,33 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
   public void doPost(HttpServletRequest request, HttpServletResponse response)
       throws ServletException, IOException {
 
-    Long userID = auth.whichUserIsLoggedIn(request, response);
+    // get parameters
+    TaskData task = (TaskData) request.getAttribute("task");
+    long taskID = Long.parseLong(request.getParameter("taskID-" + task.getName()));
```

Comment:
Try not to key the request params by the task Name, since the name is not guaranteed to be unique.  Try to use the taskID instead.